src/main/java/gov/cabinetoffice/gapuserservice/service/user/OneLoginUserService.java

```diff
@@ -474,11 +475,38 @@ public boolean hasEmailChanged(final User user, final OneLoginUserInfoDto userIn
     }
 
     public List<UserEmailDto> getUserEmailsBySubs(List<String> subs) {
-        List<User> users = userRepository.findBySubIn(subs);
-        return users.stream().map(user -> UserEmailDto.builder()
-                .emailAddress(awsEncryptionService.encryptField(user.getEmailAddress()))
-                .sub(user.getSub())
-                .build())
+
+        // Splits the list of subs into two lists. One containing OneLogin subs and the other containing COLA subs.
```

Comment:
`List<User> users = new ArrayList<>();

        List<String> oneLoginSubs = subs.stream()
                .filter(sub -> sub.contains(ONE_LOGIN_PREFIX))
                .toList();
        if (!oneLoginSubs.isEmpty()) {
            users.addAll(userRepository.findBySubIn(oneLoginSubs));
        }

        List<UUID> colaSubs = subs.stream()
                .filter(sub -> !sub.contains(ONE_LOGIN_PREFIX))
                .map(UUID::fromString)
                .toList();
        if (!colaSubs.isEmpty()) {
            users.addAll(userRepository.findByColaSubIn(colaSubs));
        }

        return users.stream()
                .map(user -> UserEmailDto.builder()
                        .emailAddress(awsEncryptionService.encryptField(user.getEmailAddress()))
                        .sub(user.getSub())
                        .build()
                )
                .toList();`
                
                Thinking something like this looks cleaner? i.e using filter instead of having the Map structure